lib/controllers/messages.ts

```diff
@@ -276,21 +257,24 @@ async function getMessageNotificationStatuses(
     // collect the statuses of all channels
     const channelStatusesPromises = Object.keys(NotificationChannelEnum)
       .map(k => NotificationChannelEnum[k as NotificationChannelEnum])
-      .map(channel => ({
+      .map(async channel => ({
         channel,
-        status: getChannelStatus(channel)
+        status: await getChannelStatus(channel)
```

Comment:
I believe you don't want an await here, the map is synchronous and creates an array of promises, the await should happen just on Promise.all

Reply:
see for instance [here](https://www.typescriptlang.org/play/#src=function%20f()%3A%20Promise%3Cstring%3E%20%7B%0A%20%20return%20Promise.resolve(%22%22)%3B%0A%7D%0A%0Aconst%20x%20%3D%20%5B%22a%22%5D.map(async%20e%20%3D%3E%20await%20f())%0A%0Aconst%20y%20%3D%20%5B%22a%22%5D.map(e%20%3D%3E%20f()))

Reply:
This was a patch to your PR as TS didn't compile complaining that there was an "unnecessary await on non promise value" (in the caller function, not here). This was because you cannot call Promise.all() on the results of this map() because the Promise is nested inside the returned object: you don't get an Array of Promises but an Array of { channel, Promise<status> }.

Reply:
ops yes sorry my mistake, yes code is good, thanks